pkgs/by-name/si/similarity/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "similarity";
+  version = "0.2.4";
+
+  src = fetchFromGitHub {
+    owner = "mizchi";
+    repo = "similarity";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-Z2ZaKBpq7N8KIX8nOzPhm8evfoUxBzaAK0+4cU9qBDE=";
+  };
+
+  cargoHash = "sha256-oYqdCHGY6OZSbYXhjIt20ZL2JkZP7UEOhn0fhuZQnZo=";
+
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  versionCheckProgram = "${placeholder "out"}/bin/${finalAttrs.pname}-ts";
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Code similarity detection tool";
+    homepage = "https://github.com/mizchi/similarity";
+    changelog = "https://github.com/mizchi/similarity/blob/${finalAttrs.version}/CHANGELOG.md";
```

Comment:
The link is wrong:
The `v` is missing from before finalAttrs.version.
(You could also use finalAttrs.src.rev or finalAttrs.src.tag directly)
I don't really care which one you go with.

Reply:
Also, sidenote: it doesn't seem like the CHANGELOG.md file is being updated, most of the changes are only listed in the release notes...